language-design.md

```diff
@@ -27,7 +98,26 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 **Response**
 
+The overarching theme we identified in _Growing a Language_ is that users should be able to extend programming languages. The keys to the theme are that programming languages should be user-oriented and user-defined. While the base of the programming language, in Bloch's words the _pattern_, of the programming language must be designed so that user programs and libraries are natural extensions of the base. The base that language designers provide should be just large enough for users to build and small enough for the program designers to be able to focus on their _pattern_. By enabling users to grow programming languages, they become contributers to the language. The size of the team working on the language is as large as the community and the people writing the language are the same as the ones who are using it. When we worked on the lab, we found ourselves asking how it would be easiest for us to use the language. That is, our design philosphy was that we, as programmers, are designing for like-minded programmers with the caveat that we hope others can use the library without reading into the basecode as much as we did. We want other users to be able to add to the library with the minimum amount of work, or conceptual weight. We redesigned the library so that the existing language would intuitive to build upon. We wanted to make it easy for users of the library to make changes to sound as easily, or intuitively as possible, but also be able to extend the language to include their own methods and objects with ease. The connection we made between growing the language and the redesign of the library is that we shared the goals of making additions to the library as _intuitive_ and _natural_. Users and programmers should be able to contribute and add upon what we build so that their contributions will become a natural extension of what we built. We think a large part of that has to do with the pattern that we establish with the methods that we wrote. We established a pattern that uses a sound object with methods that alter the sound object and separated what we felt were over-restrictive methods. In the newly designed class, users can do more specific actions to manipulate the sound. We hope that this control will enable them to write easier to understand code. We found the old code problematic since we could not easily discern the actions of the computer when calling methods. Future extensions of the class can easily follow the pattern that we developed to enable sound objects to transformed in other ways, and the extended class will seem like an updated version of the original.
```

Comment:
Firstly, I would have appreciated if this were broken up line by line, instead of using line wrapping.  It makes commenting hard :).

Secondly, I think I'd go a step further and say that a user should be able to use the library without reading into the source code _at all_.  While reading source code can be a useful exercise, and you'll certainly learn a lot about what happens under the hood, ideally it works well enough that under the hood _doesn't matter_.